Typescript definitions for runtime

seveves opened the pull request:
I've successfully tested the offline-plugin typings with 'npm pack' in a local project.
Now you can use f.e. these lines of code with typescript:
```import * as Runtime from 'offline-plugin/runtime';

Runtime.install({
  onUpdateReady: () => Runtime.applyUpdate(),
	onUpdated: () => location.reload()
});```

seveves commented:
I've updated the PR. Now you can import all interfaces through ```import { Runtime, InstallOptions } from 'offline-plugin';``` for better tooling integration (like interface documentation etc.).

NekR commented:
@seveves thanks for doing this, but let's clarify some things first because I didn't work with TypeScript for a while and not sure how declaration syntax changed.

So, first is this:
```ts
import OfflinePlugin from 'offline-plugin';
```
Is syntax for the webpack config if you write it in TypeScript, this construction shouldn't be used in bundles/entries.

```ts
import OfflinePluginRuntime from 'offline-plugin/runtime'
```

This is how it should be done in bundles/entries. 

What confuses me is this:

```ts
import { Runtime, InstallOptions } from 'offline-plugin';
```

Because this code literally should fail if it would be raw ES modules because there is no `Runtime`  export in `offline-plugin`. Also it's weird to import from `offline-plugin` directly in bundles/entries when it should be from 'offline-plugin/runtime'. Maybe it redirect imports somehow it declaration, but it should be separate I believe.

NekR commented on offline-plugin.d.ts:
```diff
@@ -0,0 +1,85 @@
+  }
+}
+
+declare var Runtime: OfflinePlugin.Runtime;
```
Shouldn't it be `OfflinePluginRuntime` interface? I believe `OfflinePlugin` and `OfflinePluginRuntime` interfaces/namespaces should be separate since they are unrelated and being used in 2 completely separate places (config vs entry).

NekR commented on README.md:
````diff
@@ -51,6 +51,22 @@ Then, add the [runtime](docs/runtime.md) into your entry file (typically main en
 require('offline-plugin/runtime').install();
 ```
 
+... and for Typescript users:
````
I think this should be a link to `docs/typescript.md`, e.g. "For TypeScript users see here"

seveves commented:
Thanks for the review. I'm on it :)

seveves commented:
I removed the offline-plugin module and added everything to 'offline-plugin/runtime'. I also removed the Runtime interface. I think exporting the functions directly is way better and easier to understand. 
Another thing I've changed is in the package.json. The official typescript docs say that it is "types" and not "typings".

We could think about adding the typedefs to the [DefinitelyTyped](http://definitelytyped.org/) project one day :)

NekR commented on offline-plugin.d.ts:
```diff
@@ -0,0 +1,73 @@
+   * 
+   * @memberOf RuntimeStatic
+   */
+ export function install(options?: InstallOptions): void;
```
Missing 2 spaces I believe.

seveves replied:
omg! no don't merge it: I'll fix this!!!

NekR commented:
@seveves hey, thank you! Looks great expect those 2 spaces (😂). I'll merge it.

NekR commented:
> We could think about adding the typedefs to the DefinitelyTyped project one day :)

If people will need. Since TypeScript 2 people use `@types/*` npm namespace, so probably better to add there. Not sure how to do that though.
